### DISCUSSION TITLE
Thousands of 'failed to generate include insertion edits' error messages in stderr log

### DISCUSSION THREAD

**Discussion by vincentjgoh** [clangd/clangd#1831] [General]
```
I[16:30:53.899] Failed to generate include insertion edits for adding header (FileURI='file:///C:/Program%20Files%20%28x86%29/Windows%20Kits/10/Include/10.0.18362.0/um/winuser.h', IncludeHeader='file:///C:/Program%20Files%20%28x86%29/Windows%20Kits/10/Include/10.0.18362.0/um/winuser.h') into d:\p4v\Dev\Foopty\Source\Foopty\s\FooptyLogic.h: Header not on include path
I[16:30:53.899] Failed to generate include insertion edits for adding header (FileURI='file:///C:/Program%20Files%20%28x86%29/Windows%20Kits/10/Include/10.0.18362.0/um/winuser.h', IncludeHeader='file:///C:/Program%20Files%20%28x86%29/Windows%20Kits/10/Include/10.0.18362.0/um/winuser.h') into d:\p4v\Dev\Foopty\Source\Foopty\s\FooptyLogic.cpp: Header not on include path
I[16:30:53.899] Failed to generate include insertion edits for adding header (FileURI='file:///C:/Program%20Files%20%28x86%29/Windows%20Kits/10/Include/10.0.18362.0/um/winnt.h', IncludeHeader='file:///C:/Program%20Files%20%28x86%29/Windows%20Kits/10/Include/10.0.18362.0/um/winnt.h') into d:\p4v\Dev\Foopty\Source\Foopty\s\FooptyLogic.h: Header not on include path
I[16:30:53.899] Failed to generate include insertion edits for adding header (FileURI='file:///C:/Program%20Files%20%28x86%29/Windows%20Kits/10/Include/10.0.18362.0/um/winuser.h', IncludeHeader='file:///C:/Program%20Files%20%28x86%29/Windows%20Kits/10/Include/10.0.18362.0/um/winuser.h') into d:\p4v\Dev\Foopty\Source\Foopty\s\FooptyLogic.cpp: Header not on include path
I[16:30:53.900] Failed to generate include insertion edits for adding header (FileURI='file:///C:/Program%20Files%20%28x86%29/Windows%20Kits/10/Include/10.0.18362.0/um/winuser.h', IncludeHeader='file:///C:/Program%20Files%20%28x86%29/Windows%20Kits/10/Include/10.0.18362.0/um/winuser.h') into d:\p4v\Dev\Foopty\Source\Foopty\s\FooptyLogic.h: Header not on include path
```

Can anyone give me some clarity on what this error message is supposed to be telling me? 

I'm working inside the FooptyLogic.cpp and FooptyLogic.h files, and I don't need winuser.h or any of the literally thousands of other suggestions that are getting spat out here. This error tends to coincide with the moment that clangd stops offering completions or symbol lookups, though restarting the LSP-mode (emacs) workspace seems to work. 

Interestingly, this never happened to me with clangd 14, and now that I've tried to swap to clangd 17, I get this error several times a day. (I've tried switching back to clang 14 and now I have the same problem. I'm going to delete the entire background cache and start fresh and see if that does anything.)

I did see this discussion that was opened ages ago and came to no meaningful conclusion: https://github.com/clangd/clangd/issues/320

I'm not sure how related it is except for the part where completions and lookups stop.

**Comment by HighCommander4** [comment]
> Can anyone give me some clarity on what this error message is supposed to be telling me?

The message is saying that when processing a code completion request, clangd determined that one of the proposed symbols is declared in (for example) `winuser.h`, and as part of accepting that proposal, clangd should insert an include of `winuser.h` at the top of the file, but clangd couldn't find a way to formulate that include directive in a way that would make the include resolve against one of the file's include paths.

The error should have no effect beyond the "add an include directive" edit being dropped from that particular completion proposal.
